### PR TITLE
CPMS post check update

### PR DESCRIPTION
**Pull request by jinyunma**
1.  merge step `cucushift-installer-check-control-plane-machine-name` into `cucushift-installer-check-control-plane-machineset` since we only need to check controlplane machine name on cluster that supports CPMS.
2.  remove step `cucushift-installer-check-control-plane-machine-name` from chain `cucushift-installer-check`
3.  Add step `cucushift-installer-check-control-plane-machineset` into chain `cucushift-installer-check`

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh [hunk 1]
```diff
@@ -42,7 +45,6 @@ function hasCPMS() {
         ;;
     esac    
 
-    version=$(oc adm release info ${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
     echo "OCP version: ${version}"
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh:48]: suggest to move this line together out of this function.

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-gcp

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-aws

File: ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh [hunk 2]
```diff
@@ -79,3 +82,35 @@ fi
...
+echo "excepted_count: ${excepted_count}"
+
+if (( ${excepted_count} < 1 )) || (( ${control_plane_nodes_count} < 1 )); then
+    echo "ERROR: control plane nodes count or expected nodes count is less than 1, exit now."
+    exit 1
+fi
+
+if [[ "${excepted_count}" != "${control_plane_nodes_count}" ]]; then
+    echo "ERROR: One or more control plane machine name is not expected."
+    exit 1
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh:110]: From the [log](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/43094/rehearse-43094-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure/1699683248948383744):
```
control_plane_nodes_count: 4
excepted_count: 1
ERROR: One or more control plane machine name is not expected.
```

I think we should add the following lines to some prior place for easy debugging.
```
echo "Machines:"
cat "${machine_output}"
```

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
@jianlinliu @gpei please help to review again, thanks, also cc @yunjiang29 @jianli-wei

**Comment by jianlinliu** [comment]
@jinyunma the azure job failure is caused by a bug?

**Comment by jinyunma** [comment]
>  the azure job failure is caused by a bug?
yes, this is known issue tracked in https://issues.redhat.com/browse/OCPBUGS-18113.

**Comment by jianlinliu** [comment]
> > the azure job failure is caused by a bug?
> 
> yes, this is known issue tracked in https://issues.redhat.com/browse/OCPBUGS-18113.

Then once we merged this PR, all the azure testing would be blocked?

File: ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh [hunk 3]
```diff
@@ -21,6 +21,10 @@ echo "OCP version: ${version}"
 
 # check if controlplanemachinesets is supported by the IaaS and the OCP version
 # return 0 if controlplanemachinesets is supported, otherwise 1
+
+version=$(oc adm release info ${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh:25]: duplicated with the line 19

File: ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh [hunk 4]
```diff
@@ -79,5 +88,41 @@ else
...
+fi
+
+if [[ "${excepted_count}" != "${control_plane_nodes_count}" ]]; then
+    echo "ERROR: One or more control plane machine name is not expected."
+    check_result=1
+else
+    echo "INFO: All control plane machine names are expected."
+fi
+
+if [[ ${check_result} == 1 ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh:123]: How about directly `exit $check_result` ?

File: ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh (hunk 3 above)
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh]: fixed

File: ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh (hunk 4 above)
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/control-plane-machinesets/cucushift-installer-check-control-plane-machinesets-commands.sh]: updated

**Pull request merged**